src/java/org/apache/cassandra/config/Config.java

```diff
@@ -216,35 +248,39 @@
-    public volatile int batch_size_fail_threshold_in_kb = 50;
+    @Replaces(oldName = "column_index_size_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public volatile DataStorage column_index_size = new DataStorage("64KB");
+    @Replaces(oldName = "column_index_cache_size_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public DataStorage column_index_cache_size = new DataStorage("2KB");
+    @Replaces(oldName = "batch_size_warn_threshold_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public DataStorage batch_size_warn_threshold = new DataStorage("5KB");
+    @Replaces(oldName = "batch_size_fail_threshold_in_kb", converter = Converter.KilobytesDataStorageConverter.class, deprecated = true)
+    public volatile DataStorage batch_size_fail_threshold = new DataStorage("50KB");
     public Integer unlogged_batch_across_partitions_warn_threshold = 10;
     public volatile Integer concurrent_compactors;
-    public volatile int compaction_throughput_mb_per_sec = 16;
-    public volatile int compaction_large_partition_warning_threshold_mb = 100;
-    public int min_free_space_per_drive_in_mb = 50;
+    @Replaces(oldName = "compaction_throughput_mb_per_sec", converter = Converter.MegabitsPerSecondBitRateConverter.class, deprecated = true)
+    public volatile BitRate compaction_throughput = new BitRate("16MiB/s");
+    @Replaces(oldName = "compaction_large_partition_warning_threshold_mb", converter = Converter.MegabytesDataStorageConverter.class, deprecated = true)
+    public volatile DataStorage compaction_large_partition_warning_threshold = new DataStorage("100MB");
+    @Replaces(oldName = "min_free_space_per_drive_in_mb", converter = Converter.MegabytesDataStorageConverter.class, deprecated = true)
+    public DataStorage min_free_space_per_drive = new DataStorage("50MB");
     public volatile Integer compaction_tombstone_warning_threshold = 100000;
 
     public volatile int concurrent_materialized_view_builders = 1;
     public volatile int reject_repair_compaction_threshold = Integer.MAX_VALUE;
-
-    /**
-     * @deprecated retry support removed on CASSANDRA-10992
-     */
-    @Deprecated
-    public int max_streaming_retries = 3;
-
-    public volatile int stream_throughput_outbound_megabits_per_sec = 200;
-    public volatile int inter_dc_stream_throughput_outbound_megabits_per_sec = 200;
+    
+    @Replaces(oldName = "stream_throughput_outbound_megabits_per_sec", converter = Converter.MegabitsPerSecondBitRateConverter.class, deprecated = true)
+    public volatile BitRate stream_throughput_outbound = new BitRate("200MiB/s");
+    @Replaces(oldName = "inter_dc_stream_throughput_outbound_megabits_per_sec", converter = Converter.MegabitsPerSecondBitRateConverter.class, deprecated = true)
+    public volatile BitRate inter_dc_stream_throughput_outbound = new BitRate("200MiB/s");
```

Comment:
These two continue to use `MegabitsPerSecondBitRateConverter` (which would actually convert from Mbps), but still `200 Mbps != 200MiB/s`, so just have to fix up that part.

Reply:
Now one question we need to think about is the conversion ratio. 200/25=8 and this is not the ratio between the mebibyte and megabits actually. But that is what we advertise currently in cassandra.yaml. 
I just made it precise calculation but I will change also the cassandra.yaml file when we confirm that this is OK.